packages/zoe/src/contractSupport/durability.js

```diff
@@ -10,7 +10,7 @@ import { Far } from '@endo/marshal';
 
 /**
  * @template {{}} E Ephemeral state
- * @param {() => E} init
+ * @param {(key: any) => E} init
```

Comment:
took me a minute to figure out why you want key here too. seems handy but could use more documentation.

```suggestion
 * @template {{}} [K=any] key on which to provision
 * @param {(key?: K) => E} init can condition on the provisioning key
```
And of course the WeakMap and provider functions get `K` too.